tutorials/basic-topics/sorting/ShellSort.md

```diff
@@ -0,0 +1,439 @@
+---
+title: 'Shell Sort'
+description: 'Sorting algorithm using a gap concept'
+hide_table_of_contents: false
+keywords:
+  - leetcode
+  - tutorial
+  - sorting
+  - algorithm
+  - shell sort
+---
+
+<TutorialAuthors names="@<Saishreekouda>"/>
+
+## Overview
+
+Shell Sort is a sorting algorithm that employs a unique gap-based strategy to improve the efficiency of the sorting process. It extends the idea from insertion sort and offers a solution with better time complexity.
+
+## Algorithm
+
+Shell Sort, also known as diminishing increment sort, operates on the principle of repeatedly sorting subarrays with a specific gap size. It gradually reduces this gap size until it becomes 1, effectively sorting the entire array.
+
+### Step 1: Choose a Gap Sequence
+
+The first crucial step in Shell Sort is selecting an appropriate gap sequence. A common choice is to start with the array's length divided by 2 and repeatedly divide by 2 until the gap becomes 1. The choice of the gap sequence greatly influences the algorithm's performance.
+
+### Step 2: Start Sorting with a Gap
+
+1. Divide the array into subarrays of size equal to the chosen gap. Initially, these subarrays consist of elements separated by the selected gap.
+
+2. For each subarray, execute an Insertion Sort. This involves rearranging elements within the subarray to their correct positions.
+
+### Step 3: Reduce the Gap
+
+3. Decrease the gap (usually by dividing it by 2).
+
+### Step 4: Repeat Until Gap is 1
+
+4. Repeat steps 2 and 3 until the gap becomes 1.
+
+### Step 5: Final Pass
+
+5. Perform a final pass of Insertion Sort with a gap of 1. This is a regular Insertion Sort that ensures the entire array is sorted correctly.
+
+Now, let's visualize how Shell Sort works with a simple example:
+
+<!-- We can see that Shell Sort is performed in the following way: -->
+
+Suppose we have an unsorted array `[6, 5, 3, 1, 8, 7, 2, 4]`. We'll perform Shell Sort on this array step by step:
+
+![image](../../../static/img/ShellSort.png) 
```

Comment:
1. please check out CONTRIBUTION guide to see how to upload image.
2. please cut the images to put the corresponding part after each step

Reply:
Okay, I will make these changes